src/test/java/net/mcreator/integration/generator/GTProcedureBlocks.java

```diff
@@ -223,6 +268,25 @@ public static void runTest(Logger LOG, String generatorName, Random random, Work
 				}
 			}
 
+			if (procedureBlock.getRepeatingStatements() != null) {
+				try {
+					JsonArray args0 = procedureBlock.blocklyJSON.getAsJsonObject().get("args0").getAsJsonArray();
+					for (StatementInput statement : procedureBlock.getRepeatingStatements()) {
+						for (int i = 0; i < args0.size(); i++) {

```

Comment:
Not sure if args0 is the way to go with statement inputs, as size of args0 does not tell much about the number of accepted statements

Reply:
I believe it does - as long as we assume that tests can get procedure blocks with any number of repeating statements _PREdefined_.

Reply:
So should I add a field type check or let normal statements loop do this loop's job?

Reply:
Not sure I fully understood the question

Reply:
How do repeating inputs then define input type? Is it even possible to define such block using JSON? Otherwise we may not actually be able to run tests on those, or we will have to require JSON to specify type of repeating inputs so we know what type to provide in the inputs. Then one could generate say 3-5 inputs/statements for each repeating one

Reply:
Well, I thought of repeating inputs/statements as a way to group several predefined/mutator-created inputs or statements so that we don't have to continuously check if an input with suffix of certain index exists but process an array of such inputs instead. At this point I need to either **a) cut the lists of repeating inputs/statements down to names only** or **b) move definition of these objects to a separate class**. Should I go **a)** or **b)**?

Reply:
Code as it is is ok, except tests; you just need to add a way to define what input type repeating inputs take so tests know what data to provide in tests

Reply:
I suggested 2 ways I thought of... 🤦‍♂️

Reply:
It was not explained well as I don't understand what a) means at all. How would cutting list to names only help with input type? Also, my suggestion has nothing to do with statement inputs as statements don't have input type..

Please improve wording on what you meant

Reply:
If you meant making new class type for inputs, then the answer is yes. Extend inputs in a way that you can also define input type so we know which input type to provide in tests.

For statement list input, all is good implementation and tests, just input list inputs need type specifier for tests

Reply:
EDIT N: sorry, my bad. I think it is ok as long as the block provides some toolbox_init as in this case logic as for normal inputs can be used. I think this is ok. I will revisit this after you fix  https://github.com/MCreator/MCreator/pull/2759#discussion_r1000894705

Reply:
>I think this is ok if the same is done here.

The tests now check if inputs (in `mcreator/inputs`) are configured by the toolbox XML. If all of them are, tests continue their checks. If at least one is not configured but is named `one_of_repeating_inputs_name + index`, tests check if all the inputs (in `args\d`) named from a repeating input are configured there and fails if not.
>>For statement list input, all is good implementation and tests, just input list inputs need type specifier for tests

The second half of the sentence contradicts what you said before this quote, which statement do I respect?

Reply:
I just wanted to cite that statement inputs are ok, disregard the second part of the quotation. I will take another look on this PR when I get time for it.

Is demo plugin now fixed?

Reply:
I needed to figure the exact feature problem to fix to know what should be fixed with the plugin (special repeating input class required me to modify demo PB definition, now I need to revert that change).

Reply:
Ok, mark as ready for review when all is ready. Thanks!